src/Handler/CurlFactory.php

```diff
@@ -161,7 +161,7 @@ private static function createRejection(EasyHandle $easy, array $ctx)
 
         // If an exception was encountered during the onHeaders event, then
         // return a rejected promise that wraps that exception.
-        if ($easy->onHeadersException) {
+        if ($easy->onHeadersException instanceof \Exception) {
```

Comment:
In a way this is also a BC...

If someone passed on `onHeadersException` a `StreamInterface` it would fail on the next line.. but now it wont.. and you will silently not see the error.

Reply:
Maybe I must revert this as well? WDYT?

Reply:
I hesitated on this line. Let’s be safe rather then sorry. 

Revert it.